### PR TITLE
Introduce ResourceList components and implement on Dashboards index

### PR DESCRIPTION
**Pull request by alexpaxton**
screen shot 2019-02-25 at 1 53 45 pm
screen shot 2019-02-25 at 1 48 21 pm

Introducing the `ResourceList` component family as a more flexible and responsive alternative to `IndexList`. The main difference being `ResourceList` does not use HMTL tables under the hood and makes use of optional render props to ensure consistency.

This PR implements `ResourceList` as a replacement for the dashboards index view.

Make all `testID` props consistent cross the UI
Update e2e test for dashboards index
Split off e2e test for dashboards view

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

**Comment by alexpaxton** [comment]
Currently there are 2 components `EditableDescription` and `EditableName` that are used in the index list views for resources. I have created `ResourceList.Name` and `ResourceList.Description` as better alternatives (also more tightly coupled to the `ResourceList` family). Once all the instances of `IndexList` are replaced with `ResourceList` we can delete `EditableDescription` and `EditableName` in favor of the new components

File: ui/src/dashboards/components/dashboard_index/DashboardCard.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+export default class DashboardCard extends PureComponent<Props> {
+  public render() {
+    const {dashboard} = this.props
+    const {id} = dashboard
+
+    return (
+      <ResourceList.Card
+        key={`dashboard-id--${id}`}
+        testID="resource-card"
+        name={() => (
```
Comment (chnn) [inline ui/src/dashboards/components/dashboard_index/DashboardCard.tsx]: This is super cool

Comment (alexpaxton) [reply ui/src/dashboards/components/dashboard_index/DashboardCard.tsx]: Pat yourself on the back

File: ui/src/clockface/components/resource_list/ResourceCard.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+// Constants
+import {UPDATED_AT_TIME_FORMAT} from 'src/dashboards/constants'
+
+interface PassedProps {
+  name: () => JSX.Element
+  description?: () => JSX.Element
+  updatedAt?: string
+  owner?: Organization
+  labels?: () => JSX.Element
+  meta1?: () => JSX.Element
```
Comment (mavarius) [inline ui/src/clockface/components/resource_list/ResourceCard.tsx]: Are these for resource metadata? The meta1 & meta2 props are ambiguous and limiting.

File: ui/src/clockface/components/resource_list/ResourceName.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+type Props = PassedProps & DefaultProps
+
+interface State {
+  isEditing: boolean
+  workingName: string
+}
+
+@ErrorHandling
+class ResourceName extends Component<Props, State> {
```
Comment (mavarius) [inline ui/src/clockface/components/resource_list/ResourceName.tsx]: Maybe we should just create generic inline edit components for this that aren't tied to the resource list. That way we can use them for name, description, and whatever else we need and they can be used outside of this list.

Comment (alexpaxton) [reply ui/src/clockface/components/resource_list/ResourceName.tsx]: I made an issue for this: https://github.com/influxdata/influxdb/issues/12171

File: ui/src/clockface/components/resource_list/ResourceCard.tsx (hunk 2 above)
Comment (alexpaxton) [reply ui/src/clockface/components/resource_list/ResourceCard.tsx]: Refactored to be a single prop `metaData: () => JSX.Element[]`

**Pull request merged**